spec/lang/well-formed.md

```diff
@@ -759,6 +776,15 @@ impl Program {
             }
         }
 
+        // Check vtables: All vtables for the same trait must have the same method names defined.
+        for (_name, vtable) in self.vtables {
+            let Some(trait_methods) = self.traits.get(vtable.trait_name) else {
+                throw_ill_formed!("Program: vtable for unknown trait");
+            };
+            let methods = vtable.methods.keys().collect::<Set<_>>();
+            ensure_wf(methods == trait_methods, "Program: vtables which defines invalid methods")?;
```

Comment:
```suggestion
            ensure_wf(methods == trait_methods, "Program: vtable has not the right set of methods")?;
```
Also, is there a test hitting this case?

Reply:
No, this PR contains neither UB nor Ill-formedness tests. I should probably add this though.

Reply:
Yes please do :)